### PR TITLE
Try to fix wrong range of events to process error

### PR DESCRIPTION
**Pull request by wkrzemien**
Change the logic in the JPetTaskIO, also add a bunch of tests in JPetInputHandler.
This is supposed to fix the problem with the event range, but I am not sure if this totally fix the problem.

### CODE REVIEW

**Comment by wkrzemien** [comment]
retest it please

**Comment by wkrzemien** [comment]
I still must fix the remaing problem but if you have any more remarks, please let me know.

File: Core/JPetTaskIO/JPetTaskIO.cpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ bool JPetTaskIO::run(const JPetDataInterface&)
         if (isProgressBar(fParams.getOptions())) {
```
Comment (grey277) [inline Core/JPetTaskIO/JPetTaskIO.cpp]: I have comment to line 95-97, if there was error in init function of subtask, do we want to continue with another tasks not terminate whole analysis? I think it is pointless to continue as next task is taking data from previous one. If It is a case, I think getEventRange, run() and terminate() should also use continue not returning false.

File: Core/JPetTaskIO/JPetTaskIO.cpp [hunk 2]
```diff
@@ -124,6 +124,7 @@ bool JPetTaskIO::run(const JPetDataInterface&)
             return false;
           }
         }
+        fInputHandler->nextEntry();
       }
     } else {
       JPetDataInterface dummyEvent;
```
Comment (grey277) [inline Core/JPetTaskIO/JPetTaskIO.cpp]: I think after line 137 we should terminate analysis if we detect error in terminate function of subtask.

File: Core/JPetTaskIO/JPetTaskIO.cpp [hunk 3]
```diff
@@ -113,7 +113,7 @@ bool JPetTaskIO::run(const JPetDataInterface&)
         if (isProgressBar(fParams.getOptions())) {
           displayProgressBar(subTaskName, i, lastEvent);
         }
-        JPetData event(fInputHandler->getNextEntry());
+        JPetData event(fInputHandler->getEntry());
         ok = pTask->run(event);
         if (!ok) {
           ERROR("In run() of:" + subTaskName + ". ");
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply Core/JPetTaskIO/JPetTaskIO.cpp]: ok. Added

File: Core/JPetTaskIO/JPetInputHandler.cpp [hunk 4]
```diff
@@ -67,6 +56,40 @@ void JPetInputHandler::closeInput()
...
+  std::tie(isOK, firstEntry, lastEntry) = calculateEntryRange(options);
+  if (!isOK) {
+    ERROR("Some error occured in getEntryRange");
+    return false;
+  }
+  fEntryRange.firstEntry = firstEntry;
+  fEntryRange.lastEntry = lastEntry;
+  fEntryRange.currentEntry = firstEntry;
+  assert (fReader);
+  fReader->nthEntry(fEntryRange.currentEntry);
```
Comment (grey277) [inline Core/JPetTaskIO/JPetInputHandler.cpp]: missing return true

File: Core/JPetTaskIO/JPetTaskIO.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (grey277) [reply Core/JPetTaskIO/JPetTaskIO.cpp]: I'm not sure if it was intended, but I do not see any return/continue/break after this error, even thought you said it was added.

**Comment by grey277** [comment]
I have small comment to line 119 of JPetTaskIO, that was marked as resolved. I'm not sure if it was intended, but I do not see any return/continue/break after this error, even thought you said it was added.

File: Core/JPetTaskIO/JPetTaskIO.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (wkrzemien) [reply Core/JPetTaskIO/JPetTaskIO.cpp]: No, I have ovewritten it. Thanks. I am adding the fix now

File: Core/JPetTaskIO/JPetInputHandler.cpp (hunk 4 above)
Comment (wkrzemien) [reply Core/JPetTaskIO/JPetInputHandler.cpp]: Added

**Comment by wkrzemien** [comment]
Ok. I think it is ok now @grey277

**Pull request merged**